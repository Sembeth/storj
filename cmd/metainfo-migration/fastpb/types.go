// Copyright (C) 2021 Storj Labs, Inc.
// See LICENSE for copying information.

// Package fastpb contains generated marshaling for Pointers.
package fastpb

import (
	"storj.io/common/storj"
)

//go:generate protoc -I=. --gogofaster_out=Mgoogle/protobuf/any.proto=github.com/gogo/protobuf/types,Mgoogle/protobuf/duration.proto=github.com/gogo/protobuf/types,Mgoogle/protobuf/struct.proto=github.com/gogo/protobuf/types,Mgoogle/protobuf/timestamp.proto=github.com/gogo/protobuf/types,Mgoogle/protobuf/wrappers.proto=github.com/gogo/protobuf/types,paths=source_relative:. pointerdb.proto

// NodeID is a unique node identifier.
type NodeID = storj.NodeID

// PieceID is a unique identifier for pieces.
type PieceID = storj.PieceID
