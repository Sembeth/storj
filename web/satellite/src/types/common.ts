// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

export enum SortDirection {
    ASCENDING = 1,
    DESCENDING,
}

export class PartneredSatellite {
    constructor(
        public name: string = '',
        public address: string = '',
    ) {}
}
