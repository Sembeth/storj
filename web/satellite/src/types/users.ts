// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

/**
 * Exposes all user-related functionality.
 */
export interface UsersApi {
    /**
     * Updates users full name and short name.
     *
     * @param user - contains information that should be updated
     * @throws Error
     */
    update(user: UpdatedUser): Promise<void>;
    /**
     * Fetch user.
     *
     * @returns User
     * @throws Error
     */
    get(): Promise<User>;
}

/**
 * User class holds info for User entity.
 */
export class User {
    public id: string;
    public fullName: string;
    public shortName: string;
    public email: string;
    public partner: string;
    public partnerId: string;
    public password: string;

    public constructor(id: string = '', fullName: string = '', shortName: string = '', email: string = '', partner: string = '', partnerId: string = '', password: string = '') {
        this.id = id;
        this.fullName = fullName;
        this.shortName = shortName;
        this.email = email;
        this.partner = partner;
        this.partnerId = partnerId;
        this.password = password;
    }

    public getFullName(): string {
        return this.shortName === '' ? this.fullName : this.shortName;
    }
}

/**
 * User class holds info for updating User.
 */
export class UpdatedUser {
    public fullName: string;
    public shortName: string;

    public constructor(fullName: string, shortName: string) {
        this.fullName = fullName;
        this.shortName = shortName;
    }

    public setFullName(value: string) {
        this.fullName = value.trim();
    }

    public setShortName(value: string) {
        this.shortName = value.trim();
    }

    public isValid(): boolean {
        return !!this.fullName;
    }
}
