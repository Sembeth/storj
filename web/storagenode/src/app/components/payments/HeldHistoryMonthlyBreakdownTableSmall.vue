// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div class="held-history-table-container--small__item">
        <div class="held-history-table-container--small__item__satellite-info">
            <div>
                <p class="held-history-table-container--small__item__satellite-info__name">{{ heldHistoryItem.satelliteName }}</p>
                <p class="held-history-table-container--small__item__satellite-info__months">{{ heldHistoryItem.age }} month</p>
            </div>
            <div class="held-history-table-container--small__item__satellite-info__button">
                <div class="icon hide" @click="hide" v-if="isExpanded">
                    <blue-hide-icon></blue-hide-icon>
                </div>
                <div class="icon expand" @click="expand" v-else>
                    <blue-expand-icon></blue-expand-icon>
                </div>
            </div>
        </div>
        <transition name="fade">
            <div class="held-history-table-container--small__item__held-info" v-if="isExpanded">
                <div class="held-history-table-container--small__item__held-info__item">
                    <p class="held-history-table-container--small__item__held-info__item__label">Month 1-3</p>
                    <p class="held-history-table-container--small__item__held-info__item__value">{{ heldHistoryItem.firstPeriod | centsToDollars }}</p>
                </div>
                <div class="held-history-table-container--small__item__held-info__item">
                    <p class="held-history-table-container--small__item__held-info__item__label">Month 4-6</p>
                    <p class="held-history-table-container--small__item__held-info__item__value">{{ heldHistoryItem.secondPeriod | centsToDollars }}</p>
                </div>
                <div class="held-history-table-container--small__item__held-info__item">
                    <p class="held-history-table-container--small__item__held-info__item__label">Month 7-9</p>
                    <p class="held-history-table-container--small__item__held-info__item__value">{{ heldHistoryItem.thirdPeriod | centsToDollars }}</p>
                </div>
            </div>
        </transition>
    </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

import BlueHideIcon from '@/../static/images/common/BlueMinus.svg';
import BlueExpandIcon from '@/../static/images/common/BluePlus.svg';

import { HeldHistoryMonthlyBreakdownItem } from '@/app/types/payout';

@Component({
    components: {
        BlueExpandIcon,
        BlueHideIcon,
    },
})
export default class HeldHistoryMonthlyBreakdownTableSmall extends Vue {
    @Prop({default: () => new HeldHistoryMonthlyBreakdownItem()})
    public readonly heldHistoryItem: HeldHistoryMonthlyBreakdownItem;

    /**
     * Indicates if held info should be rendered.
     */
    public isExpanded: boolean = false;

    /**
     * Shows held info.
     */
    public expand(): void {
        this.isExpanded = true;
    }

    /**
     * Hides held info.
     */
    public hide(): void {
        this.isExpanded = false;
    }
}
</script>

<style scoped lang="scss">
    .held-history-table-container--small__item {
        padding: 12px;
        width: calc(100% - 24px);

        &__satellite-info {
            display: flex;
            align-items: center;
            justify-content: space-between;

            &__name {
                font-family: 'font_regular', sans-serif;
                font-size: 14px;
                color: var(--regular-text-color);
                max-width: calc(100% - 40px);
                word-break: break-word;
            }

            &__months {
                font-family: 'font_regular', sans-serif;
                font-size: 11px;
                color: #9b9db1;
                margin-top: 3px;
            }

            &__button {
                display: flex;
                align-items: center;
                justify-content: center;
                width: 30px;
                height: 30px;
                min-width: 30px;
                min-height: 30px;
                background: var(--expand-button-background-color);
                border-radius: 3px;
                cursor: pointer;
            }
        }

        &__held-info {
            margin-top: 16px;

            &__item {
                display: flex;
                align-items: center;
                justify-content: space-between;
                font-size: 12px;
                line-height: 12px;
                margin-bottom: 10px;

                &__label {
                    font-family: 'font_medium', sans-serif;
                    color: #909bad;
                }

                &__value {
                    font-family: 'font_regular', sans-serif;
                    color: var(--regular-text-color);
                }
            }
        }
    }

    .icon {
        display: flex;
        align-items: center;
        justify-content: center;
        max-width: 100%;
        max-height: 100%;
        width: 100%;
        height: 100%;
    }

    .fade-enter-active,
    .fade-leave-active {
        transition: opacity 0.5s;
    }

    .fade-enter,
    .fade-leave-to /* .fade-leave-active below version 2.1.8 */ {
        opacity: 0;
    }
</style>
