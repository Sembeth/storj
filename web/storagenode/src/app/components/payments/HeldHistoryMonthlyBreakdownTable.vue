// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div>
        <div class="held-history-table-container--large">
            <div class="held-history-table-container--large__labels-area">
                <div class="column justify-start column-1">
                    <p class="held-history-table-container--large__labels-area__text">Satellite</p>
                </div>
                <div class="column justify-end column-2">
                    <p class="held-history-table-container--large__labels-area__text">Month 1-3</p>
                </div>
                <div class="column justify-end column-3">
                    <p class="held-history-table-container--large__labels-area__text">Month 4-6</p>
                </div>
                <div class="column justify-end column-4">
                    <p class="held-history-table-container--large__labels-area__text">Month 7-9</p>
                </div>
            </div>
            <div v-for="item in monthlyBreakdown" class="held-history-table-container--large__info-area" :key="item.satelliteID">
                <div class="justify-start column-1">
                    <p class="held-history-table-container--large__info-area__text">{{ item.satelliteName }}</p>
                    <p class="held-history-table-container--large__info-area__months">{{ item.age }} month</p>
                </div>
                <div class="column justify-end column-2">
                    <p class="held-history-table-container--large__info-area__text">{{ item.firstPeriod | centsToDollars }}</p>
                </div>
                <div class="column justify-end column-3">
                    <p class="held-history-table-container--large__info-area__text">{{ item.secondPeriod | centsToDollars }}</p>
                </div>
                <div class="column justify-end column-4">
                    <p class="held-history-table-container--large__info-area__text">{{ item.thirdPeriod | centsToDollars }}</p>
                </div>
            </div>
        </div>
        <div class="held-history-table-container--small">
            <HeldHistoryMonthlyBreakdownTableSmall
                v-for="item in monthlyBreakdown"
                :held-history-item="item"
                :key="item.satelliteID"
            />
        </div>
    </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

import HeldHistoryMonthlyBreakdownTableSmall from '@/app/components/payments/HeldHistoryMonthlyBreakdownTableSmall.vue';

import { HeldHistoryMonthlyBreakdownItem } from '@/app/types/payout';

@Component({
    components: {
        HeldHistoryMonthlyBreakdownTableSmall,
    },
})
export default class HeldHistoryMonthlyBreakdownTable extends Vue {
    /**
     * Returns list of satellite held history items by periods from store.
     */
    public get monthlyBreakdown(): HeldHistoryMonthlyBreakdownItem[] {
        return this.$store.state.payoutModule.heldHistory.monthlyBreakdown;
    }
}
</script>

<style scoped lang="scss">
    .held-history-table-container--large {

        &__labels-area {
            display: flex;
            flex-direction: row;
            align-items: center;
            justify-content: space-between;
            margin-top: 17px;
            padding: 0 16px;
            width: calc(100% - 32px);
            height: 36px;
            background: var(--table-header-color);

            &__text {
                font-family: 'font_medium', sans-serif;
                font-size: 14px;
                color: #909bad;
            }
        }

        &__info-area {
            padding: 11px 16px;
            display: flex;
            flex-direction: row;
            align-items: center;
            justify-content: space-between;
            min-height: 34px;
            height: auto;
            border-bottom: 1px solid rgba(169, 181, 193, 0.3);

            &:last-of-type {
                border-bottom: none;
            }

            &__text {
                font-family: 'font_regular', sans-serif;
                font-size: 14px;
                color: var(--regular-text-color);
                max-width: 100%;
                word-break: break-word;
            }

            &__months {
                font-family: 'font_regular', sans-serif;
                font-size: 11px;
                color: #9b9db1;
                margin-top: 3px;
            }
        }
    }

    .held-history-table-container--small {
        display: none;
    }

    .column {
        display: flex;
        flex-direction: row;
        align-items: center;
    }

    .justify-start {
        justify-content: flex-start;
    }

    .justify-end {
        justify-content: flex-end;
    }

    .column-1 {
        width: 37%;
    }

    .column-2,
    .column-3,
    .column-4 {
        width: 21%;
    }

    @media screen and (max-width: 600px) {

        .held-history-table-container--large {
            display: none;
        }

        .held-history-table-container--small {
            display: block;
        }
    }
</style>
