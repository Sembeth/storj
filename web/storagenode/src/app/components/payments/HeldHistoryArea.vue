// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <section class="held-history-container">
        <div class="held-history-container__header">
            <p class="held-history-container__header__title">Held Amount History</p>
        </div>
        <div class="held-history-container__divider"></div>
        <HeldHistoryMonthlyBreakdownTable />
    </section>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

import HeldHistoryMonthlyBreakdownTable from '@/app/components/payments/HeldHistoryMonthlyBreakdownTable.vue';

import { PAYOUT_ACTIONS } from '@/app/store/modules/payout';

@Component({
    components: {
        HeldHistoryMonthlyBreakdownTable,
    },
})
export default class HeldHistoryArea extends Vue {
    /**
     * Lifecycle hook before component render.
     * Fetches held history information.
     */
    public beforeMount(): void {
        this.$store.dispatch(PAYOUT_ACTIONS.GET_HELD_HISTORY);
    }
}
</script>

<style scoped lang="scss">
    .held-history-container {
        display: flex;
        flex-direction: column;
        padding: 28px 40px 10px 40px;
        background: var(--block-background-color);
        border: 1px solid var(--block-border-color);
        box-sizing: border-box;
        border-radius: 12px;
        margin: 12px 0 50px;

        &__header {
            display: flex;
            flex-direction: row;
            align-items: center;
            justify-content: space-between;

            &__title {
                font-family: 'font_medium', sans-serif;
                font-size: 18px;
                color: var(--regular-text-color);
            }
        }

        &__divider {
            width: 100%;
            height: 1px;
            margin-top: 18px;
            background-color: #eaeaea;
        }
    }
</style>
