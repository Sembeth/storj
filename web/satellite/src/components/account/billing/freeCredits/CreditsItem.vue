// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div class="container">
        <p class="container__item">{{ startDate }}</p>
        <p class="container__item coupon">{{ creditType }}</p>
        <p :class="{'expired' : expirationCheck}" class="container__item">{{ expiration }}</p>
        <p class="container__item">{{  creditsItem.amount | centsToDollars }}</p>
        <p class="container__item">{{ amountUsed | centsToDollars }}</p>
        <p class="container__item available">{{ creditsItem.remaining | centsToDollars }}</p>
    </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

import PaymentsHistoryItemDate from '@/components/account/billing/depositAndBillingHistory/PaymentsHistoryItemDate.vue';

import { PaymentsHistoryItem, PaymentsHistoryItemType } from '@/types/payments';
import { MONTHS_NAMES } from '@/utils/constants/date';

@Component({
    components: {
        PaymentsHistoryItemDate,
    },
})
export default class CreditsItem extends Vue {
    @Prop({default: () => new PaymentsHistoryItem()})
    private readonly creditsItem: PaymentsHistoryItem;

    /**
     * Return credit type string depending on item type.
     */
    public get creditType(): string {
        const trial = 'Free Tier Credit';

        if (this.creditsItem.type === PaymentsHistoryItemType.Coupon) {
            return trial;
        }

        return '';
    }

    /**
     * Returns memory amount depending on item's money amount.
     */
    public get memoryAmount(): number {
        const gbPrice: number = 5.5; // in cents.

        return Math.floor(this.creditsItem.amount / gbPrice);
    }

    /**
     * Returns formatted string of expiration date.
     */
    public get expiration(): string {
        if (!this.creditsItem.hasExpiration) {
            return 'Never expires';
        }

        const monthNumber = this.creditsItem.end.getUTCMonth();
        const year = this.creditsItem.end.getUTCFullYear();

        return `${MONTHS_NAMES[monthNumber]} ${year}`;
    }

    /**
     * Returns formatted string of start date.
     */
    public get startDate(): string {
        const monthNumber = this.creditsItem.start.getUTCMonth();
        const year = this.creditsItem.start.getUTCFullYear();

        return `${MONTHS_NAMES[monthNumber]} ${year}`;
    }

    /**
     * Returns remaining amount
     */
    public get amountUsed(): number {
        const amount = this.creditsItem.amount;
        const remaining = this.creditsItem.remaining;

        return amount - remaining;
    }

    /**
     * Checks for coupon expiration.
     */
    public get expirationCheck(): boolean {
        return this.creditsItem.hasExpiration && this.creditsItem.end.getTime() < new Date().getTime();
    }
}
</script>

<style scoped lang="scss">
    .container {
        display: flex;
        width: 100%;

        &__item {
            min-width: 16.6%;
            font-family: 'font_regular', sans-serif;
            text-align: left;
            font-size: 16px;
            line-height: 19px;
            color: #354049;
        }

        &__item.coupon {
            font-family: 'font_bold', sans-serif;
        }

        &__item.expired {
            color: #ce3030;
        }
    }

</style>
