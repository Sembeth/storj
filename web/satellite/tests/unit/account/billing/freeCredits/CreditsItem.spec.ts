// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import CreditsItem from '@/components/account/billing/freeCredits/CreditsItem.vue';

import { PaymentsHistoryItem, PaymentsHistoryItemType } from '@/types/payments';
import { createLocalVue, mount } from '@vue/test-utils';

const localVue = createLocalVue();
const couponActive = new PaymentsHistoryItem('testId', 'desc', 275, 0, 'Active', '', new Date(1), new Date(1), PaymentsHistoryItemType.Coupon, 275);
const couponExpired = new PaymentsHistoryItem('testId', 'desc', 275, 0, 'Expired', '', new Date(1), new Date(1), PaymentsHistoryItemType.Coupon, 0);
const couponUsed = new PaymentsHistoryItem('testId', 'desc', 500, 0, 'Used', '', new Date(1), new Date(1), PaymentsHistoryItemType.Coupon, 0);

localVue.filter('centsToDollars', (cents: number): string => {
    return `$${(cents / 100).toFixed(2)}`;
});

describe('CreditsItem', (): void => {
    it('renders correctly if not expired', (): void => {
        const wrapper = mount(CreditsItem, {
            localVue,
            propsData: {
                creditsItem: couponActive,
            },
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly if expired', (): void => {
        const wrapper = mount(CreditsItem, {
            localVue,
            propsData: {
                creditsItem: couponExpired,
            },
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly if used', (): void => {
        const wrapper = mount(CreditsItem, {
            localVue,
            propsData: {
                creditsItem: couponUsed,
            },
        });

        expect(wrapper).toMatchSnapshot();
    });
});
