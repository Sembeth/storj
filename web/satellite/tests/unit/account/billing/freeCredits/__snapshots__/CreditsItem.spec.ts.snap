// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreditsItem renders correctly 1`] = `
<div class="container">
  <p class="container__item">January 1970</p>
  <p class="container__item coupon">Free Tier Credit</p>
  <p class="container__item expired">January 1970</p>
  <p class="container__item">$2.75</p>
  <p class="container__item">$0.00</p>
  <p class="container__item available">$2.75</p>
</div>
`;
